### PR TITLE
prevent profile view of expensify notification email

### PR DESCRIPTION
**Pull request by mkhutornyi**
### Fixed Issues
$ https://github.com/Expensify/App/issues/32500
PROPOSAL: https://github.com/Expensify/App/issues/32500#issuecomment-1841370563

### Tests
Same as QA step

- [ ] Verify that no errors appear in the JS console

### Offline tests
Same as QA step

### QA Steps
Case 1:
1. Open New Expensify app
2. Navigate to any conversation
3. Request money from another account by uploading a non-legible receipt (a random pic).
4. Wait for smartscan to finish. It should fail.
5. Navigate to the IOU details
6. Create a thread with a system message about a receipt scanning failure
7. In the thread header, hover on the "Expensify" avatar
8. Verify that notifications@expensify.com doesn't show in tooltip
9. Click on avatar
10. In the RHP that appears, click on "Members"
11. In "Members," click on "Expensify" (verify that subtitle is "Hidden")
12. Verify that "Expensify" is not clickable

Case 2:
1. Open New Expensify app
2. Navigate to any conversation
3. Send any message containing https://staging.new.expensify.com/a/11665625 (In dev test: https://dev.new.expensify.com:8082/a/11665625)
4. Click on that link
5. Verify that not found page shows

Case 3:
1. Open New Expensify app
2. Navigate to any conversation
3. Try to mention @notifications@expensify.com
4. Verify that this email doesn't show in mentions list

- [ ] Verify that no errors appear in the JS console

### PR Author Checklist
- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
        - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

https://github.com/Expensify/App/assets/97676131/bb69481c-1b94-4ca3-890e-a659338e07f0

</details>

<details>
<summary>Android: mWeb Chrome</summary>

https://github.com/Expensify/App/assets/97676131/5cf3b18d-4be9-4f11-b40e-b34c8f3fdce0

</details>

<details>
<summary>iOS: Native</summary>

https://github.com/Expensify/App/assets/97676131/3bce3f0d-611b-4afd-9129-7ad39b2ecfe2

</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/Expensify/App/assets/97676131/ca4a1878-ad1d-4eb1-b58b-45b50a45a9b1

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/Expensify/App/assets/97676131/bc172039-7d43-4a71-901b-1d555bfd71f1

</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/Expensify/App/assets/97676131/76ac701c-8ba5-4807-9dc5-f565c54684ca

</details>

### CODE REVIEW

File: src/pages/ReportParticipantsPage.js [hunk 1]
```diff
@@ -88,7 +88,7 @@ function ReportParticipantsPage(props) {
     const styles = useThemeStyles();
     const participants = _.map(getAllParticipants(props.report, props.personalDetails, props.translate), (participant) => ({
         ...participant,
-        isDisabled: ReportUtils.isOptimisticPersonalDetail(participant.accountID),
+        isDisabled: ReportUtils.isOptimisticPersonalDetail(participant.accountID) || CONST.RESTRICTED_ACCOUNT_IDS.includes(participant.accountID),
```
[the remark this answers is not in the document]
Comment (blimpich) [reply src/pages/ReportParticipantsPage.js]: Lets hide it from the mention list. We should be consistent in our UI. The app shouldn't suggest a bot account is just like any other user that can be messaged, so hiding it in the mention list and also hiding the email in the tooltip is perfectly acceptable for this I think.

**Review by fedirjh — CHANGES_REQUESTED** [review]
Looks good, left some comments about the code.

**Review by fedirjh — APPROVED** [review]
Looks good to me

**Comment by blimpich** [comment]
Will review these changes tomorrow morning. It's late for me.

File: src/CONST.ts [hunk 2]
```diff
@@ -1556,6 +1556,15 @@ const CONST = {
         ];
     },
 
+    // Emails that profile view is prohibited
+    get RESTRICTED_EMAILS() {
+        return [this.EMAIL.NOTIFICATIONS] as string[];
```
Comment (blimpich) [inline src/CONST.ts:1561]: Please don't typecast. It defeats the purpose of using TypeScript. Instead just add a return type.

```
    get RESTRICTED_EMAILS(): readonly string[] {
        return [this.EMAIL.NOTIFICATIONS];
    },
```

File: src/pages/ReportParticipantsPage.tsx [hunk 3]
```diff
@@ -76,7 +76,7 @@ function ReportParticipantsPage({report, personalDetails}: ReportParticipantsPag
 
     const participants = getAllParticipants(report, personalDetails, translate).map((participant) => ({
         ...participant,
-        isDisabled: participant?.accountID ? ReportUtils.isOptimisticPersonalDetail(participant.accountID) : false,
```
[the remark this answers is not in the document]
Comment (blimpich) [reply src/pages/ReportParticipantsPage.tsx:79]: Do we know why they changed it? Feel free to ask them

**Review by blimpich — CHANGES_REQUESTED** [review]

File: src/pages/ReportParticipantsPage.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (blimpich) [reply src/pages/ReportParticipantsPage.tsx:79]: Sweet thank you 🙂. Lets see if we get an answer there and then I think we can merge if nothing comes up from that.

[the remark this answers is not in the document]
Comment (blimpich) [reply src/pages/ReportParticipantsPage.tsx:79]: Alright, lets not hold this up any longer on their response. @mkhutornyi can you fix the merge conflict and then I'll approve and merge?

**Review by blimpich — APPROVED** [review]

**Pull request merged**